src/components/templates/Poster/usePosterVideo.ts

```diff
@@ -0,0 +1,132 @@
+import { useEffect, useMemo, useState } from "react";
+import Video, { Room, LocalParticipant, RemoteParticipant } from "twilio-video";
+import firebase from "firebase/app";
+
+import { withId, WithId } from "utils/id";
+
+import { User } from "types/User";
+
+import { useUser } from "hooks/useUser";
+import { useWorldUsersById } from "hooks/users";
+
+// TODO: create useVideoRoom hook and divide this into two hooks & refactor it
+export const usePosterVideo = (venueId: string) => {
+  const [room, setRoom] = useState<Room>();
+  const [participants, setParticipants] = useState<
+    Array<LocalParticipant | RemoteParticipant>
+  >([]);
+  const [token, setToken] = useState<string>();
+  const [hasVideo, setHasVideo] = useState(false);
+
+  const { userId } = useUser();
+  const { worldUsersById } = useWorldUsersById();
```

Comment:
You should use `useWorldUsersByIdWorkaround` as it works around typing deficiencies.

Reply:
Done in https://github.com/sparkletown/sparkle/pull/1299/commits/1c71872238b0cf77414c943211ee5416cc65a456